utam-core/src/main/java/utam/core/driver/Navigation.java

```diff
@@ -44,4 +47,46 @@ public interface Navigation {
    */
   void closeWindow();
 
+  /**
+   * Gets the number of currently opened managed windows
+   *
+   * @return the number of currently opened managed windows
+   */
+  int getWindowCount();
+
+  /**
+   * Gets the window with the current command context
+   *
+   * @return the window with the current command context
+   */
+  Window currentWindow();
+
+  /**
+   * Sets up a wait for a new window to appear after taking a subsequent action
+   */
+  void setupWaitForNewWindow();
+
+  /**
+   * Waits for a new window to be opened and switches the command context to the new window;
+   * if called without a prior call to setupWaitForNewWindow, an exception is thrown
+   *
+   * @return the new window
+   */
+  Window waitForNewWindow();
+
+  /**
+   * Waits for a new window to be opened, switches to it, and loads a Page Object that is a new root
+   *
+   * @return a Page Object that is a new root
+   */
+  PageObject waitForNewWindowAndLoad(Class<? extends RootPageObject> pageObjectType);
```

Comment:
The signature here should be:

    public <T extends RootPageObject> T waitForNewWindowAndLoad(Class<T> type)

Don't forget to update the JavaDoc when you've made the change.